pallets/ddc-customers/src/lib.rs

```diff
@@ -244,55 +244,30 @@ pub mod pallet {
 
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
-		/// Create new bucket with provided public availability & reserved resources
+		/// Create new bucket with specified cluster id
 		///
 		/// Anyone can create a bucket
 		#[pallet::weight(10_000)]
-		pub fn create_bucket(
-			origin: OriginFor<T>,
-			public_availability: bool,
-			resources_reserved: u128,
-		) -> DispatchResult {
+		pub fn create_bucket(origin: OriginFor<T>, cluster_id: ClusterId) -> DispatchResult {
 			let bucket_owner = ensure_signed(origin)?;
 			let cur_bucket_id = Self::buckets_count();
 
+			ensure!(
+				ddc_clusters::pallet::Clusters::<T>::contains_key(&cluster_id),
+				Error::<T>::ClusterDoesNotExist
+			);
+
 			let bucket = Bucket {
 				bucket_id: cur_bucket_id + 1,
 				owner_id: bucket_owner,
-				cluster_id: None,
-				public_availability,
-				resources_reserved,
+				cluster_id: Some(cluster_id),
```

Comment:
Let's make the `cluster_id` field non-optional as now every bucket must be assigned to some cluster